modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/coercion/NumericComparisonExecutionTest.java

```diff
@@ -0,0 +1,993 @@
+        }
+
+        Arguments fail() {
+            opMatcher = ofBoolType(false);
+
+            return Arguments.of(pair, expression, opMatcher);
+        }
+
+        Arguments ok() {
+            opMatcher = ofBoolType(true);
+
+            return Arguments.of(pair, expression, opMatcher);
+        }
+
+        Arguments type(NumericPair type, Operation op) {
+            opMatcher = ofType(type, op);
+
+            return Arguments.of(pair, expression, opMatcher);
+        }
+    }
+
+    private static Matcher<Object> ofBoolType(Boolean compResult) {
+        return new BaseMatcher<>() {
+            Object actual;
+
+            @Override
+            public boolean matches(Object actual) {
+                assert actual != null;
+                Pair<Object, ColumnMetadata> pair = (Pair<Object, ColumnMetadata>) actual;
+                this.actual = pair.getFirst();
+                return compResult.equals(this.actual);
+            }
+
+            @Override
+            public void describeTo(Description description) {
+                description.appendText(format("Expected : '{}' but found '{}'", compResult, actual));
+            }
+        };
+    }
+
+    private static Matcher<Object> ofType(NumericPair typesPair, Operation op) {
+        return new BaseMatcher<>() {
+            Object actual;
+            ClassInfoHolder innerClassInfo;
+            int precision = 0;
+            int scale = 0;
+            ColumnMetadata colMeta;
+
+            @Override
+            public boolean matches(Object actual) {
+                assert actual != null;
+                Pair<Object, ColumnMetadata> pair = (Pair<Object, ColumnMetadata>) actual;
+                this.actual = pair.getFirst();
+                colMeta = pair.getSecond();
+
+                boolean checkPrecisionScale = false;
+
+                if (this.actual instanceof BigDecimal) {
+                    precision = ((BigDecimal) this.actual).precision();
+                    scale = ((BigDecimal) this.actual).scale();
+                    checkPrecisionScale = true;
+                }
+
+                innerClassInfo = TYPES_MAPPING.get(op).get(typesPair);
+
+                boolean precCheck = checkPrecisionScale ? colMeta.precision() >= precision && colMeta.scale() >= scale : true;
+
+                // negative scale in meta for operations with integer and real\double types.
+                return precCheck && innerClassInfo.clazz.isInstance(this.actual)
+                        && innerClassInfo.precision == precision
+                        && innerClassInfo.scale == scale;
+            }
+
+            @Override
+            public void describeTo(Description description) {
+                description.appendText(format("Expected : '{}' but found '{}, precision: {}, scale: {}', column meta: {}",
+                        innerClassInfo, actual.getClass(), precision, scale, colMeta));
+            }
+        };
+    }
+
+    static {
```

Comment:
it would be nice to write test cases in the same style as it is for planner tests

```
private static Stream<Arguments> sumArgs() {
        return Stream.of(
                forTypePair(NumericPair.TINYINT_TINYINT)
                        .resultWillBe(ofType(NativeType.INT8))),
                ...
        );
    }
```